pkg/driver/node/node.go

```diff
@@ -115,6 +118,18 @@ func (ns *S3NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePubl
 	}
 
 	args := mountpoint.ParseArgs(mountpointArgs)
+
+	if capMount := volCap.GetMount(); capMount != nil && util.UsePodMounter() {
+		if volumeMountGroup := capMount.GetVolumeMountGroup(); capMount.GetVolumeMountGroup() != "" {
```

Comment:
Regarding my first comment, I was thinking of adding `args.Has(mountpoint.ArgGid)` check here. 
```suggestion
		if volumeMountGroup := capMount.GetVolumeMountGroup(); volumeMountGroup != "" && !args.Has(mountpoint.ArgGid) {
```

So we don't end up updating arguments partially if the user already provided a `gid` (and probably other settings)